### PR TITLE
Publish to Maven Central

### PR DESCRIPTION
**Pull request by jmewes**

### CODE REVIEW

**Comment by jmewes** [comment]
**Current benefits:**

In the previous WIP pull request, the `publishToSonatype` Gradle task was successfully executed. Also the signing worked.

See https://github.com/ePages-de/restdocs-wiremock/pull/72#issuecomment-796837414

**Current limitations:**

- The documentation in the README is not yet updated. I want to do this once the library is actually available on Maven Central.
- The release needs to be closed and released in the Sonatype UI. It could be automated but it is recommended by the "publish-plugin" README to start with the manual process.
- The `publishToSonatype` was executed successfully but created no staging repo yet, because this will only happen if the task is run on a commit with a tag.

**Next steps:**

Once this PR is merged I want to create a release with the version number "0.8.0". This should trigger a Travis build and create staging repository at Sonatype. This I want to manually close and release which should make `restdocs-wiremock` available on Maven Central.

File: ci_publish.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+if [[ "$TRAVIS_TAG" =~ ^[0-9.]+ ]] ; then
+	./gradlew publishToSonatype \
+		--exclude-task :restdocs-server:publishToSonatype \
+		--info \
+		-Dorg.gradle.project.sonatypeUsername="${SONATYPE_USERNAME}" \
+		-Dorg.gradle.project.sonatypePassword="${SONATYPE_PASSWORD}" \
+	 	-Dorg.gradle.project.signing.keyId="${SIGNING_KEY_ID}" \
+		-Dorg.gradle.project.signing.password="${SIGNING_PASSWORD}" \
+		-Dorg.gradle.project.signing.secretKeyRingFile="${SIGNING_KEYRING_FILE}"
```
Comment (ozscheyge) [inline ci_publish.sh:12]: Nice solution to inject the environment variables!

File: README.md [hunk 2]
```diff
@@ -359,17 +373,37 @@ Please execute at least step 1 + 2 if before importing restdocs-wiremock into yo
...
+./gradlew currentVersion
+
+# (a) Release new patch version
+./gradlew release -Prelease.versionIncrementer=incrementPatch
+
+# (b) Release new minor version
+./gradlew release -Prelease.versionIncrementer=incrementMajor
+
+## (c) Release new major version
+./gradlew release -Prelease.versionIncrementer=incrementMajor
```
Comment (ozscheyge) [inline README.md:392]: > Once this PR is merged I want to create a release with the version number "0.8.0". This should trigger a Travis build and create staging repository at Sonatype. This I want to manually close and release which should make restdocs-wiremock available on Maven Central.

Do you intend to create the release using GitHub UI? If so and if this works, I hope those steps shouldn't be necessary.

**Review by ozscheyge — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (jmewes) [reply README.md:392]: No, as far as I understand calling this Gradle task will also create the release entry on GitHub.

Let's see how it works. We can later discuss the pros and cons of using this Gradle plugin.

Comment (jmewes) [reply README.md:392]: I had the following problems with the `release` task.

1. I mixed up the terms "major" version and "minor" version (both in the description in the README file and when actually calling the task)
2. The plugin also tries to update the version number in the README file. So I cannot create a release independently due to the branch protection
3. The plugin tries to push to the `origin` repository which points to my fork. So some additional configuration is needed.

All in all, there are many things one needs to understand when using the tool. So I am now also convinced that a handful of manual steps for creating a release are better than using this plugin.

**Pull request merged**